contractcourt/breach_resolver.go

```diff
@@ -0,0 +1,99 @@
+package contractcourt
+
+import (
+	"encoding/binary"
+	"io"
+
+	"github.com/lightningnetwork/lnd/channeldb"
+)
+
+// breachResolver is a resolver that will handle breached closes. In the
+// future, this will likely take over the duties the current breacharbiter has.
+type breachResolver struct {
+	// resolved reflects if the contract has been fully resolved or not.
+	resolved bool
+
+	contractResolverKit
+}
+
+// newBreachResolver instantiates a new breach resolver.
+func newBreachResolver(resCfg ResolverConfig) *breachResolver {
+	r := &breachResolver{
+		contractResolverKit: *newContractResolverKit(resCfg),
+	}
+
+	r.initLogger(r)
+
+	return r
+}
+
+// ResolverKey returns the unique identifier for this resolver.
+func (b *breachResolver) ResolverKey() []byte {
+	key := newResolverID(b.ChanPoint)
+	return key[:]
+}
+
+// Resolve queries the breacharbiter to see if the justice transaction has been
+// broadcast.
+func (b *breachResolver) Resolve() (ContractResolver, error) {
+	breached, err := b.IsBreached(&b.ChanPoint)
```

Comment:
I would expect this to just block until the resolver has finished. This method is usually called in a goroutine.

Reply:
So a for loop with a quit statement to match the other ones?

Reply:
Yeah, so it blocks fully until the breach arbiter is done. Otherwise as is, this'll cause a tight loop here: https://github.com/lightningnetwork/lnd/blob/master/contractcourt/channel_arbitrator.go#L2195

Other resolvers wait for things like spends/confirmations to complete before returning here.